src/main/java/com/minecraftabnormals/endergetic/core/mixin/PlayerListMixin.java

```diff
@@ -40,7 +53,19 @@ private void removeBalloons(ServerPlayerEntity player, CallbackInfo info) {
 	@Inject(at = @At("RETURN"), method = "initializeConnectionToPlayer")
 	private void spawnBalloons(NetworkManager netManager, ServerPlayerEntity player, CallbackInfo info) {
 		ServerWorld serverWorld = (ServerWorld) player.world;
-		CompoundNBT compound = ((PlayerList) (Object) this).readPlayerDataFromFile(player);
+
+		CompoundNBT compound = this.server.getServerConfiguration().getHostPlayerNBT();
+		if (!(compound != null && player.getName().getString().equals(this.server.getServerOwner()))) {
+			try {
+				File file1 = new File(this.playerDataManager.getPlayerDataFolder(), player.getCachedUniqueIdString() + ".dat");
```

Comment:
`file1` could be changed to a more descriptive name.
For example, `playerDataFile` would work nicely.